numpy/core/include/numpy/ufuncobject.h

```diff
@@ -209,6 +209,19 @@ typedef struct _tagPyUFuncObject {
          * set by nditer object.
          */
         npy_uint32 iter_flags;
+
+        /* New in version 1 and above */
+
+        /*
+         * sizes of `frozen` core dimensions, -1 if unset (not frozen)
+         */
+        npy_intp *core_dim_szs;
+
+        /*
+         * for each core_num_dim_ix, 1 for flexible (signature has ?) 0 otherwise
+         */
+        int *core_dim_flexible;
```

Comment:
Implementation suggestion: I think we might as well make this a set of flags, so that we can re-use them for future expansions.

In fact, one could then add not just a flag `UFUNC_COREDIM_FLEXIBLE' but also a `UFUNC_COREDIM_FIXED' - for that case, `core_num_dim_idx` could just be set to the size, since there is no need to track the index.

Reply:
I'm going ahead and try this - I guess one can see it as my contribution to the BIDS sprint!